{
 "cells": [
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Troubleshooting VEP Output and Filtering**\n",
    "\n",
    "The VEP filtering procedure performed before came up with much less variants than expected for the entire human proteome. We need to find out some reason why this could be the case."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/y3/mxnsyg_910x11q_t3z54wkbr0000gn/T/ipykernel_33917/3535052536.py:13: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  vep_df = pd.read_csv(VEP_OUTPUT_PATH, sep='\\t')\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import sys\n",
    "import pandas as pd\n",
    "from IPython.display import display\n",
    "\n",
    "ROOT_DIR = os.environ.get('PYTHONPATH')\n",
    "\n",
    "TROUBLESHOOT_PATH = os.path.join(ROOT_DIR, 'files/vep_troubleshoot')\n",
    "\n",
    "VEP_OUTPUT_PATH = os.path.join(TROUBLESHOOT_PATH, 'uniprot_human_variants_vep_all.csv')\n",
    "\n",
    "# keep dataframe in memory\n",
    "vep_df = pd.read_csv(VEP_OUTPUT_PATH, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>SWISSPROT</th>\n",
       "      <th>REF_ALLELE</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>CANONICAL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NC_000002.12:g.88978974G&gt;A</td>\n",
       "      <td>A0A075B6H7.48</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>4</td>\n",
       "      <td>P/L</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NC_000002.12:g.88978975G&gt;T</td>\n",
       "      <td>A0A075B6H7.48</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>4</td>\n",
       "      <td>P/T</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NC_000002.12:g.88978971G&gt;A</td>\n",
       "      <td>A0A075B6H7.48</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>5</td>\n",
       "      <td>A/V</td>\n",
       "      <td>-</td>\n",
       "      <td>2.033e-05</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NC_000002.12:g.88978972C&gt;T</td>\n",
       "      <td>A0A075B6H7.48</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>5</td>\n",
       "      <td>A/T</td>\n",
       "      <td>-</td>\n",
       "      <td>6.774e-06</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NC_000002.12:g.88978968T&gt;G</td>\n",
       "      <td>A0A075B6H7.48</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>6</td>\n",
       "      <td>Q/P</td>\n",
       "      <td>-</td>\n",
       "      <td>6.773e-06</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681953</th>\n",
       "      <td>NC_000001.11:g.158999996T&gt;C</td>\n",
       "      <td>W6CW81.33</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>106</td>\n",
       "      <td>H/R</td>\n",
       "      <td>-</td>\n",
       "      <td>2.63e-05</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681954</th>\n",
       "      <td>NC_000001.11:g.158999987G&gt;C</td>\n",
       "      <td>W6CW81.33</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>109</td>\n",
       "      <td>S/C</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681955</th>\n",
       "      <td>NC_000001.11:g.158999987G&gt;T</td>\n",
       "      <td>W6CW81.33</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>109</td>\n",
       "      <td>S/Y</td>\n",
       "      <td>-</td>\n",
       "      <td>1.314e-05</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681956</th>\n",
       "      <td>NC_000001.11:g.158999978G&gt;A</td>\n",
       "      <td>W6CW81.33</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>112</td>\n",
       "      <td>A/V</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681957</th>\n",
       "      <td>NC_000001.11:g.158999976A&gt;C</td>\n",
       "      <td>W6CW81.33</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>113</td>\n",
       "      <td>C/G</td>\n",
       "      <td>-</td>\n",
       "      <td>6.572e-06</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10681958 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  #Uploaded_variation      SWISSPROT REF_ALLELE Allele   \n",
       "0          NC_000002.12:g.88978974G>A  A0A075B6H7.48          G      A  \\\n",
       "1          NC_000002.12:g.88978975G>T  A0A075B6H7.48          G      T   \n",
       "2          NC_000002.12:g.88978971G>A  A0A075B6H7.48          G      A   \n",
       "3          NC_000002.12:g.88978972C>T  A0A075B6H7.48          C      T   \n",
       "4          NC_000002.12:g.88978968T>G  A0A075B6H7.48          T      G   \n",
       "...                               ...            ...        ...    ...   \n",
       "10681953  NC_000001.11:g.158999996T>C      W6CW81.33          T      C   \n",
       "10681954  NC_000001.11:g.158999987G>C      W6CW81.33          G      C   \n",
       "10681955  NC_000001.11:g.158999987G>T      W6CW81.33          G      T   \n",
       "10681956  NC_000001.11:g.158999978G>A      W6CW81.33          G      A   \n",
       "10681957  NC_000001.11:g.158999976A>C      W6CW81.33          A      C   \n",
       "\n",
       "               Consequence Protein_position Amino_acids CLIN_SIG gnomADg_AF   \n",
       "0         missense_variant                4         P/L        -          -  \\\n",
       "1         missense_variant                4         P/T        -          -   \n",
       "2         missense_variant                5         A/V        -  2.033e-05   \n",
       "3         missense_variant                5         A/T        -  6.774e-06   \n",
       "4         missense_variant                6         Q/P        -  6.773e-06   \n",
       "...                    ...              ...         ...      ...        ...   \n",
       "10681953  missense_variant              106         H/R        -   2.63e-05   \n",
       "10681954  missense_variant              109         S/C        -          -   \n",
       "10681955  missense_variant              109         S/Y        -  1.314e-05   \n",
       "10681956  missense_variant              112         A/V        -          -   \n",
       "10681957  missense_variant              113         C/G        -  6.572e-06   \n",
       "\n",
       "         CANONICAL  \n",
       "0              YES  \n",
       "1              YES  \n",
       "2              YES  \n",
       "3              YES  \n",
       "4              YES  \n",
       "...            ...  \n",
       "10681953       YES  \n",
       "10681954       YES  \n",
       "10681955       YES  \n",
       "10681956       YES  \n",
       "10681957       YES  \n",
       "\n",
       "[10681958 rows x 10 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "display(vep_df)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Try to get rows with benign/pathogenic disregarding ambiguity**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/y3/mxnsyg_910x11q_t3z54wkbr0000gn/T/ipykernel_33917/1594980902.py:5: UserWarning: This pattern is interpreted as a regular expression, and has match groups. To actually get the groups, use str.extract.\n",
      "  vep_all_benign_df = vep_df[vep_df['CLIN_SIG'].str.contains(benignRegex, regex=True) & vep_df['Consequence'].str.contains(consequenceRegex, regex=True)]\n",
      "/var/folders/y3/mxnsyg_910x11q_t3z54wkbr0000gn/T/ipykernel_33917/1594980902.py:6: UserWarning: This pattern is interpreted as a regular expression, and has match groups. To actually get the groups, use str.extract.\n",
      "  vep_all_pathogenic_df = vep_df[vep_df['CLIN_SIG'].str.contains(pathogenicRegex, regex=True) & vep_df['Consequence'].str.contains(consequenceRegex, regex=True)]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>SWISSPROT</th>\n",
       "      <th>REF_ALLELE</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>CANONICAL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12190</th>\n",
       "      <td>NC_000002.12:g.210023010T&gt;C</td>\n",
       "      <td>A0AUZ9.110</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>968</td>\n",
       "      <td>D/G</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.0006836</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12192</th>\n",
       "      <td>NC_000002.12:g.210023010T&gt;C</td>\n",
       "      <td>A0AUZ9.110</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>968</td>\n",
       "      <td>D/G</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.0006836</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15535</th>\n",
       "      <td>NC_000011.10:g.19234818A&gt;G</td>\n",
       "      <td>A0AVK6.137</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>231</td>\n",
       "      <td>I/T</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.0001445</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15536</th>\n",
       "      <td>NC_000011.10:g.19234818A&gt;G</td>\n",
       "      <td>A0AVK6.137</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>231</td>\n",
       "      <td>I/T</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.0001445</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15623</th>\n",
       "      <td>NC_000011.10:g.19230828C&gt;G</td>\n",
       "      <td>A0AVK6.137</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>358</td>\n",
       "      <td>S/T</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.0122</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10680783</th>\n",
       "      <td>NC_000010.11:g.119898888C&gt;T</td>\n",
       "      <td>Q9Y6Y8.175</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>209</td>\n",
       "      <td>P/S</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.008413</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681168</th>\n",
       "      <td>NC_000010.11:g.119926067C&gt;T</td>\n",
       "      <td>Q9Y6Y8.175</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>718</td>\n",
       "      <td>A/V</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.002959</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681169</th>\n",
       "      <td>NC_000010.11:g.119926067C&gt;T</td>\n",
       "      <td>Q9Y6Y8.175</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>718</td>\n",
       "      <td>A/V</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.002959</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681650</th>\n",
       "      <td>NC_000008.11:g.119105892C&gt;T</td>\n",
       "      <td>Q9Y6Z7.155</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>179</td>\n",
       "      <td>R/W</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.001565</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681651</th>\n",
       "      <td>NC_000008.11:g.119105892C&gt;T</td>\n",
       "      <td>Q9Y6Z7.155</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>179</td>\n",
       "      <td>R/W</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.001565</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>207545 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  #Uploaded_variation   SWISSPROT REF_ALLELE Allele   \n",
       "12190     NC_000002.12:g.210023010T>C  A0AUZ9.110          T      C  \\\n",
       "12192     NC_000002.12:g.210023010T>C  A0AUZ9.110          T      C   \n",
       "15535      NC_000011.10:g.19234818A>G  A0AVK6.137          A      G   \n",
       "15536      NC_000011.10:g.19234818A>G  A0AVK6.137          A      G   \n",
       "15623      NC_000011.10:g.19230828C>G  A0AVK6.137          C      G   \n",
       "...                               ...         ...        ...    ...   \n",
       "10680783  NC_000010.11:g.119898888C>T  Q9Y6Y8.175          C      T   \n",
       "10681168  NC_000010.11:g.119926067C>T  Q9Y6Y8.175          C      T   \n",
       "10681169  NC_000010.11:g.119926067C>T  Q9Y6Y8.175          C      T   \n",
       "10681650  NC_000008.11:g.119105892C>T  Q9Y6Z7.155          C      T   \n",
       "10681651  NC_000008.11:g.119105892C>T  Q9Y6Z7.155          C      T   \n",
       "\n",
       "               Consequence Protein_position Amino_acids       CLIN_SIG   \n",
       "12190     missense_variant              968         D/G         benign  \\\n",
       "12192     missense_variant              968         D/G         benign   \n",
       "15535     missense_variant              231         I/T  likely_benign   \n",
       "15536     missense_variant              231         I/T  likely_benign   \n",
       "15623     missense_variant              358         S/T         benign   \n",
       "...                    ...              ...         ...            ...   \n",
       "10680783  missense_variant              209         P/S         benign   \n",
       "10681168  missense_variant              718         A/V         benign   \n",
       "10681169  missense_variant              718         A/V         benign   \n",
       "10681650  missense_variant              179         R/W  likely_benign   \n",
       "10681651  missense_variant              179         R/W  likely_benign   \n",
       "\n",
       "         gnomADg_AF CANONICAL  \n",
       "12190     0.0006836       YES  \n",
       "12192     0.0006836       YES  \n",
       "15535     0.0001445       YES  \n",
       "15536     0.0001445       YES  \n",
       "15623        0.0122       YES  \n",
       "...             ...       ...  \n",
       "10680783   0.008413       YES  \n",
       "10681168   0.002959       YES  \n",
       "10681169   0.002959       YES  \n",
       "10681650   0.001565       YES  \n",
       "10681651   0.001565       YES  \n",
       "\n",
       "[207545 rows x 10 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>SWISSPROT</th>\n",
       "      <th>REF_ALLELE</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>CANONICAL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9703</th>\n",
       "      <td>NC_000014.9:g.23104829G&gt;A</td>\n",
       "      <td>A0A1B0GTW7.27</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>31</td>\n",
       "      <td>S/F</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>0.001065</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9922</th>\n",
       "      <td>NC_000014.9:g.23102250G&gt;A</td>\n",
       "      <td>A0A1B0GTW7.27</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>384</td>\n",
       "      <td>S/L</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>1.971e-05</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14558</th>\n",
       "      <td>NC_000007.14:g.139161018A&gt;G</td>\n",
       "      <td>A0AVF1.135</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>263</td>\n",
       "      <td>N/S</td>\n",
       "      <td>likely_pathogenic,pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14559</th>\n",
       "      <td>NC_000007.14:g.139161018A&gt;G</td>\n",
       "      <td>A0AVF1.135</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>263</td>\n",
       "      <td>N/S</td>\n",
       "      <td>likely_pathogenic,pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22222</th>\n",
       "      <td>NC_000007.14:g.122303281G&gt;A</td>\n",
       "      <td>A0PJY2.135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>278</td>\n",
       "      <td>H/Y</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10680087</th>\n",
       "      <td>NC_000001.11:g.7677682C&gt;T</td>\n",
       "      <td>Q9Y6Y1.171</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>955</td>\n",
       "      <td>R/W</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10680187</th>\n",
       "      <td>NC_000001.11:g.7736507A&gt;G</td>\n",
       "      <td>Q9Y6Y1.171</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1077</td>\n",
       "      <td>Y/C</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10680188</th>\n",
       "      <td>NC_000001.11:g.7736507A&gt;G</td>\n",
       "      <td>Q9Y6Y1.171</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1077</td>\n",
       "      <td>Y/C</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681645</th>\n",
       "      <td>NC_000008.11:g.119105885C&gt;G</td>\n",
       "      <td>Q9Y6Z7.155</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>176</td>\n",
       "      <td>C/W</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681647</th>\n",
       "      <td>NC_000008.11:g.119105885C&gt;G</td>\n",
       "      <td>Q9Y6Z7.155</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>176</td>\n",
       "      <td>C/W</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>204963 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  #Uploaded_variation      SWISSPROT REF_ALLELE Allele   \n",
       "9703        NC_000014.9:g.23104829G>A  A0A1B0GTW7.27          G      A  \\\n",
       "9922        NC_000014.9:g.23102250G>A  A0A1B0GTW7.27          G      A   \n",
       "14558     NC_000007.14:g.139161018A>G     A0AVF1.135          A      G   \n",
       "14559     NC_000007.14:g.139161018A>G     A0AVF1.135          A      G   \n",
       "22222     NC_000007.14:g.122303281G>A     A0PJY2.135          G      A   \n",
       "...                               ...            ...        ...    ...   \n",
       "10680087    NC_000001.11:g.7677682C>T     Q9Y6Y1.171          C      T   \n",
       "10680187    NC_000001.11:g.7736507A>G     Q9Y6Y1.171          A      G   \n",
       "10680188    NC_000001.11:g.7736507A>G     Q9Y6Y1.171          A      G   \n",
       "10681645  NC_000008.11:g.119105885C>G     Q9Y6Z7.155          C      G   \n",
       "10681647  NC_000008.11:g.119105885C>G     Q9Y6Z7.155          C      G   \n",
       "\n",
       "               Consequence Protein_position Amino_acids   \n",
       "9703      missense_variant               31         S/F  \\\n",
       "9922      missense_variant              384         S/L   \n",
       "14558     missense_variant              263         N/S   \n",
       "14559     missense_variant              263         N/S   \n",
       "22222     missense_variant              278         H/Y   \n",
       "...                    ...              ...         ...   \n",
       "10680087  missense_variant              955         R/W   \n",
       "10680187  missense_variant             1077         Y/C   \n",
       "10680188  missense_variant             1077         Y/C   \n",
       "10681645  missense_variant              176         C/W   \n",
       "10681647  missense_variant              176         C/W   \n",
       "\n",
       "                              CLIN_SIG gnomADg_AF CANONICAL  \n",
       "9703                        pathogenic   0.001065       YES  \n",
       "9922                        pathogenic  1.971e-05       YES  \n",
       "14558     likely_pathogenic,pathogenic          -       YES  \n",
       "14559     likely_pathogenic,pathogenic          -       YES  \n",
       "22222                       pathogenic          -       YES  \n",
       "...                                ...        ...       ...  \n",
       "10680087             likely_pathogenic          -       YES  \n",
       "10680187             likely_pathogenic          -       YES  \n",
       "10680188             likely_pathogenic          -       YES  \n",
       "10681645                    pathogenic          -       YES  \n",
       "10681647                    pathogenic          -       YES  \n",
       "\n",
       "[204963 rows x 10 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "benignRegex = r'(benign)'\n",
    "pathogenicRegex = r'(pathogenic|damaging)'\n",
    "consequenceRegex = r'(missense_variant)'\n",
    "\n",
    "vep_all_benign_df = vep_df[vep_df['CLIN_SIG'].str.contains(benignRegex, regex=True) & vep_df['Consequence'].str.contains(consequenceRegex, regex=True)]\n",
    "vep_all_pathogenic_df = vep_df[vep_df['CLIN_SIG'].str.contains(pathogenicRegex, regex=True) & vep_df['Consequence'].str.contains(consequenceRegex, regex=True)]\n",
    "\n",
    "display(vep_all_benign_df)\n",
    "display(vep_all_pathogenic_df)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Check how many rows reduced when you exclude ambiguous mutations**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/y3/mxnsyg_910x11q_t3z54wkbr0000gn/T/ipykernel_33917/3878221960.py:5: UserWarning: This pattern is interpreted as a regular expression, and has match groups. To actually get the groups, use str.extract.\n",
      "  vep_unamb_benign_df = vep_all_benign_df[~vep_all_benign_df['CLIN_SIG'].str.contains(ambiguousRegex, regex=True)]\n",
      "/var/folders/y3/mxnsyg_910x11q_t3z54wkbr0000gn/T/ipykernel_33917/3878221960.py:6: UserWarning: This pattern is interpreted as a regular expression, and has match groups. To actually get the groups, use str.extract.\n",
      "  vep_unamb_pathogenic_df = vep_all_pathogenic_df[~vep_all_pathogenic_df['CLIN_SIG'].str.contains(ambiguousRegex, regex=True)]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>SWISSPROT</th>\n",
       "      <th>REF_ALLELE</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>CANONICAL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12190</th>\n",
       "      <td>NC_000002.12:g.210023010T&gt;C</td>\n",
       "      <td>A0AUZ9.110</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>968</td>\n",
       "      <td>D/G</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.0006836</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12192</th>\n",
       "      <td>NC_000002.12:g.210023010T&gt;C</td>\n",
       "      <td>A0AUZ9.110</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>968</td>\n",
       "      <td>D/G</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.0006836</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15535</th>\n",
       "      <td>NC_000011.10:g.19234818A&gt;G</td>\n",
       "      <td>A0AVK6.137</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>231</td>\n",
       "      <td>I/T</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.0001445</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15536</th>\n",
       "      <td>NC_000011.10:g.19234818A&gt;G</td>\n",
       "      <td>A0AVK6.137</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>231</td>\n",
       "      <td>I/T</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.0001445</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15623</th>\n",
       "      <td>NC_000011.10:g.19230828C&gt;G</td>\n",
       "      <td>A0AVK6.137</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>358</td>\n",
       "      <td>S/T</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.0122</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10680783</th>\n",
       "      <td>NC_000010.11:g.119898888C&gt;T</td>\n",
       "      <td>Q9Y6Y8.175</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>209</td>\n",
       "      <td>P/S</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.008413</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681168</th>\n",
       "      <td>NC_000010.11:g.119926067C&gt;T</td>\n",
       "      <td>Q9Y6Y8.175</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>718</td>\n",
       "      <td>A/V</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.002959</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681169</th>\n",
       "      <td>NC_000010.11:g.119926067C&gt;T</td>\n",
       "      <td>Q9Y6Y8.175</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>718</td>\n",
       "      <td>A/V</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.002959</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681650</th>\n",
       "      <td>NC_000008.11:g.119105892C&gt;T</td>\n",
       "      <td>Q9Y6Z7.155</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>179</td>\n",
       "      <td>R/W</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.001565</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681651</th>\n",
       "      <td>NC_000008.11:g.119105892C&gt;T</td>\n",
       "      <td>Q9Y6Z7.155</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>179</td>\n",
       "      <td>R/W</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.001565</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>160065 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  #Uploaded_variation   SWISSPROT REF_ALLELE Allele   \n",
       "12190     NC_000002.12:g.210023010T>C  A0AUZ9.110          T      C  \\\n",
       "12192     NC_000002.12:g.210023010T>C  A0AUZ9.110          T      C   \n",
       "15535      NC_000011.10:g.19234818A>G  A0AVK6.137          A      G   \n",
       "15536      NC_000011.10:g.19234818A>G  A0AVK6.137          A      G   \n",
       "15623      NC_000011.10:g.19230828C>G  A0AVK6.137          C      G   \n",
       "...                               ...         ...        ...    ...   \n",
       "10680783  NC_000010.11:g.119898888C>T  Q9Y6Y8.175          C      T   \n",
       "10681168  NC_000010.11:g.119926067C>T  Q9Y6Y8.175          C      T   \n",
       "10681169  NC_000010.11:g.119926067C>T  Q9Y6Y8.175          C      T   \n",
       "10681650  NC_000008.11:g.119105892C>T  Q9Y6Z7.155          C      T   \n",
       "10681651  NC_000008.11:g.119105892C>T  Q9Y6Z7.155          C      T   \n",
       "\n",
       "               Consequence Protein_position Amino_acids       CLIN_SIG   \n",
       "12190     missense_variant              968         D/G         benign  \\\n",
       "12192     missense_variant              968         D/G         benign   \n",
       "15535     missense_variant              231         I/T  likely_benign   \n",
       "15536     missense_variant              231         I/T  likely_benign   \n",
       "15623     missense_variant              358         S/T         benign   \n",
       "...                    ...              ...         ...            ...   \n",
       "10680783  missense_variant              209         P/S         benign   \n",
       "10681168  missense_variant              718         A/V         benign   \n",
       "10681169  missense_variant              718         A/V         benign   \n",
       "10681650  missense_variant              179         R/W  likely_benign   \n",
       "10681651  missense_variant              179         R/W  likely_benign   \n",
       "\n",
       "         gnomADg_AF CANONICAL  \n",
       "12190     0.0006836       YES  \n",
       "12192     0.0006836       YES  \n",
       "15535     0.0001445       YES  \n",
       "15536     0.0001445       YES  \n",
       "15623        0.0122       YES  \n",
       "...             ...       ...  \n",
       "10680783   0.008413       YES  \n",
       "10681168   0.002959       YES  \n",
       "10681169   0.002959       YES  \n",
       "10681650   0.001565       YES  \n",
       "10681651   0.001565       YES  \n",
       "\n",
       "[160065 rows x 10 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>SWISSPROT</th>\n",
       "      <th>REF_ALLELE</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>CANONICAL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9703</th>\n",
       "      <td>NC_000014.9:g.23104829G&gt;A</td>\n",
       "      <td>A0A1B0GTW7.27</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>31</td>\n",
       "      <td>S/F</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>0.001065</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9922</th>\n",
       "      <td>NC_000014.9:g.23102250G&gt;A</td>\n",
       "      <td>A0A1B0GTW7.27</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>384</td>\n",
       "      <td>S/L</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>1.971e-05</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14558</th>\n",
       "      <td>NC_000007.14:g.139161018A&gt;G</td>\n",
       "      <td>A0AVF1.135</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>263</td>\n",
       "      <td>N/S</td>\n",
       "      <td>likely_pathogenic,pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14559</th>\n",
       "      <td>NC_000007.14:g.139161018A&gt;G</td>\n",
       "      <td>A0AVF1.135</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>263</td>\n",
       "      <td>N/S</td>\n",
       "      <td>likely_pathogenic,pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22222</th>\n",
       "      <td>NC_000007.14:g.122303281G&gt;A</td>\n",
       "      <td>A0PJY2.135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>278</td>\n",
       "      <td>H/Y</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10680087</th>\n",
       "      <td>NC_000001.11:g.7677682C&gt;T</td>\n",
       "      <td>Q9Y6Y1.171</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>955</td>\n",
       "      <td>R/W</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10680187</th>\n",
       "      <td>NC_000001.11:g.7736507A&gt;G</td>\n",
       "      <td>Q9Y6Y1.171</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1077</td>\n",
       "      <td>Y/C</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10680188</th>\n",
       "      <td>NC_000001.11:g.7736507A&gt;G</td>\n",
       "      <td>Q9Y6Y1.171</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1077</td>\n",
       "      <td>Y/C</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681645</th>\n",
       "      <td>NC_000008.11:g.119105885C&gt;G</td>\n",
       "      <td>Q9Y6Z7.155</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>176</td>\n",
       "      <td>C/W</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681647</th>\n",
       "      <td>NC_000008.11:g.119105885C&gt;G</td>\n",
       "      <td>Q9Y6Z7.155</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>176</td>\n",
       "      <td>C/W</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>150399 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  #Uploaded_variation      SWISSPROT REF_ALLELE Allele   \n",
       "9703        NC_000014.9:g.23104829G>A  A0A1B0GTW7.27          G      A  \\\n",
       "9922        NC_000014.9:g.23102250G>A  A0A1B0GTW7.27          G      A   \n",
       "14558     NC_000007.14:g.139161018A>G     A0AVF1.135          A      G   \n",
       "14559     NC_000007.14:g.139161018A>G     A0AVF1.135          A      G   \n",
       "22222     NC_000007.14:g.122303281G>A     A0PJY2.135          G      A   \n",
       "...                               ...            ...        ...    ...   \n",
       "10680087    NC_000001.11:g.7677682C>T     Q9Y6Y1.171          C      T   \n",
       "10680187    NC_000001.11:g.7736507A>G     Q9Y6Y1.171          A      G   \n",
       "10680188    NC_000001.11:g.7736507A>G     Q9Y6Y1.171          A      G   \n",
       "10681645  NC_000008.11:g.119105885C>G     Q9Y6Z7.155          C      G   \n",
       "10681647  NC_000008.11:g.119105885C>G     Q9Y6Z7.155          C      G   \n",
       "\n",
       "               Consequence Protein_position Amino_acids   \n",
       "9703      missense_variant               31         S/F  \\\n",
       "9922      missense_variant              384         S/L   \n",
       "14558     missense_variant              263         N/S   \n",
       "14559     missense_variant              263         N/S   \n",
       "22222     missense_variant              278         H/Y   \n",
       "...                    ...              ...         ...   \n",
       "10680087  missense_variant              955         R/W   \n",
       "10680187  missense_variant             1077         Y/C   \n",
       "10680188  missense_variant             1077         Y/C   \n",
       "10681645  missense_variant              176         C/W   \n",
       "10681647  missense_variant              176         C/W   \n",
       "\n",
       "                              CLIN_SIG gnomADg_AF CANONICAL  \n",
       "9703                        pathogenic   0.001065       YES  \n",
       "9922                        pathogenic  1.971e-05       YES  \n",
       "14558     likely_pathogenic,pathogenic          -       YES  \n",
       "14559     likely_pathogenic,pathogenic          -       YES  \n",
       "22222                       pathogenic          -       YES  \n",
       "...                                ...        ...       ...  \n",
       "10680087             likely_pathogenic          -       YES  \n",
       "10680187             likely_pathogenic          -       YES  \n",
       "10680188             likely_pathogenic          -       YES  \n",
       "10681645                    pathogenic          -       YES  \n",
       "10681647                    pathogenic          -       YES  \n",
       "\n",
       "[150399 rows x 10 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import re\n",
    "\n",
    "ambiguousRegex = r'(uncertain|conflicting)'\n",
    "\n",
    "vep_unamb_benign_df = vep_all_benign_df[~vep_all_benign_df['CLIN_SIG'].str.contains(ambiguousRegex, regex=True)]\n",
    "vep_unamb_pathogenic_df = vep_all_pathogenic_df[~vep_all_pathogenic_df['CLIN_SIG'].str.contains(ambiguousRegex, regex=True)]\n",
    "\n",
    "display(vep_unamb_benign_df)\n",
    "display(vep_unamb_pathogenic_df)\n",
    "\n",
    "BENIGN_OUTFILE = os.path.join(TROUBLESHOOT_PATH, 'vep_benign_variants.csv')\n",
    "PATHOG_OUTFILE = os.path.join(TROUBLESHOOT_PATH, 'vep_pathogenic_variants.csv')\n",
    "\n",
    "vep_unamb_benign_df.to_csv(BENIGN_OUTFILE, sep='\\t', index=False)\n",
    "vep_unamb_pathogenic_df.to_csv(PATHOG_OUTFILE, sep='\\t', index=False)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Exclude Non-reviewed Proteins**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/y3/mxnsyg_910x11q_t3z54wkbr0000gn/T/ipykernel_33917/3256544208.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  vep_unamb_benign_df['SWISSPROT'] = std_uniprot(vep_unamb_benign_df['SWISSPROT'])\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>SWISSPROT</th>\n",
       "      <th>REF_ALLELE</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>CANONICAL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12190</th>\n",
       "      <td>NC_000002.12:g.210023010T&gt;C</td>\n",
       "      <td>A0AUZ9</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>968</td>\n",
       "      <td>D/G</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.0006836</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12192</th>\n",
       "      <td>NC_000002.12:g.210023010T&gt;C</td>\n",
       "      <td>A0AUZ9</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>968</td>\n",
       "      <td>D/G</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.0006836</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15535</th>\n",
       "      <td>NC_000011.10:g.19234818A&gt;G</td>\n",
       "      <td>A0AVK6</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>231</td>\n",
       "      <td>I/T</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.0001445</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15536</th>\n",
       "      <td>NC_000011.10:g.19234818A&gt;G</td>\n",
       "      <td>A0AVK6</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>231</td>\n",
       "      <td>I/T</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.0001445</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15623</th>\n",
       "      <td>NC_000011.10:g.19230828C&gt;G</td>\n",
       "      <td>A0AVK6</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>358</td>\n",
       "      <td>S/T</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.0122</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10680783</th>\n",
       "      <td>NC_000010.11:g.119898888C&gt;T</td>\n",
       "      <td>Q9Y6Y8</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>209</td>\n",
       "      <td>P/S</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.008413</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681168</th>\n",
       "      <td>NC_000010.11:g.119926067C&gt;T</td>\n",
       "      <td>Q9Y6Y8</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>718</td>\n",
       "      <td>A/V</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.002959</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681169</th>\n",
       "      <td>NC_000010.11:g.119926067C&gt;T</td>\n",
       "      <td>Q9Y6Y8</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>718</td>\n",
       "      <td>A/V</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.002959</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681650</th>\n",
       "      <td>NC_000008.11:g.119105892C&gt;T</td>\n",
       "      <td>Q9Y6Z7</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>179</td>\n",
       "      <td>R/W</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.001565</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681651</th>\n",
       "      <td>NC_000008.11:g.119105892C&gt;T</td>\n",
       "      <td>Q9Y6Z7</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>179</td>\n",
       "      <td>R/W</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.001565</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>160065 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  #Uploaded_variation SWISSPROT REF_ALLELE Allele   \n",
       "12190     NC_000002.12:g.210023010T>C    A0AUZ9          T      C  \\\n",
       "12192     NC_000002.12:g.210023010T>C    A0AUZ9          T      C   \n",
       "15535      NC_000011.10:g.19234818A>G    A0AVK6          A      G   \n",
       "15536      NC_000011.10:g.19234818A>G    A0AVK6          A      G   \n",
       "15623      NC_000011.10:g.19230828C>G    A0AVK6          C      G   \n",
       "...                               ...       ...        ...    ...   \n",
       "10680783  NC_000010.11:g.119898888C>T    Q9Y6Y8          C      T   \n",
       "10681168  NC_000010.11:g.119926067C>T    Q9Y6Y8          C      T   \n",
       "10681169  NC_000010.11:g.119926067C>T    Q9Y6Y8          C      T   \n",
       "10681650  NC_000008.11:g.119105892C>T    Q9Y6Z7          C      T   \n",
       "10681651  NC_000008.11:g.119105892C>T    Q9Y6Z7          C      T   \n",
       "\n",
       "               Consequence Protein_position Amino_acids       CLIN_SIG   \n",
       "12190     missense_variant              968         D/G         benign  \\\n",
       "12192     missense_variant              968         D/G         benign   \n",
       "15535     missense_variant              231         I/T  likely_benign   \n",
       "15536     missense_variant              231         I/T  likely_benign   \n",
       "15623     missense_variant              358         S/T         benign   \n",
       "...                    ...              ...         ...            ...   \n",
       "10680783  missense_variant              209         P/S         benign   \n",
       "10681168  missense_variant              718         A/V         benign   \n",
       "10681169  missense_variant              718         A/V         benign   \n",
       "10681650  missense_variant              179         R/W  likely_benign   \n",
       "10681651  missense_variant              179         R/W  likely_benign   \n",
       "\n",
       "         gnomADg_AF CANONICAL  \n",
       "12190     0.0006836       YES  \n",
       "12192     0.0006836       YES  \n",
       "15535     0.0001445       YES  \n",
       "15536     0.0001445       YES  \n",
       "15623        0.0122       YES  \n",
       "...             ...       ...  \n",
       "10680783   0.008413       YES  \n",
       "10681168   0.002959       YES  \n",
       "10681169   0.002959       YES  \n",
       "10681650   0.001565       YES  \n",
       "10681651   0.001565       YES  \n",
       "\n",
       "[160065 rows x 10 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "REVIEWED_UNIPROT_PATH = os.path.join(ROOT_DIR, 'files/uniprot_human_reviewed_lt_2700aa.csv')\n",
    "\n",
    "def std_uniprot(uniprot: str):\n",
    "\n",
    "   pattern = r'\\..*'\n",
    "   \n",
    "   return uniprot.replace(pattern, '', regex=True)\n",
    "\n",
    "# standardize SWISSPROT column\n",
    "vep_unamb_benign_df['SWISSPROT'] = std_uniprot(vep_unamb_benign_df['SWISSPROT'])\n",
    "vep_unamb_pathogenic_df['SWISSPROT'] = std_uniprot(vep_unamb_pathogenic_df['SWISSPROT'])\n",
    "\n",
    "display(vep_unamb_benign_df)\n",
    "display(vep_unamb_pathogenic_df)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
