{
 "cells": [
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Troubleshooting VEP Output and Filtering**\n",
    "\n",
    "The VEP filtering procedure performed before came up with much less variants than expected for the entire human proteome. We need to find out some reason why this could be the case."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/y3/mxnsyg_910x11q_t3z54wkbr0000gn/T/ipykernel_39382/1304747978.py:13: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  VEP_DF = pd.read_csv(VEP_OUTPUT_PATH, sep='\\t')\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import sys\n",
    "import pandas as pd\n",
    "from IPython.display import display\n",
    "\n",
    "ROOT_DIR = os.environ.get('PYTHONPATH')\n",
    "\n",
    "TROUBLESHOOT_PATH = os.path.join(ROOT_DIR, 'files/vep_troubleshoot')\n",
    "\n",
    "VEP_OUTPUT_PATH = os.path.join(TROUBLESHOOT_PATH, 'uniprot_human_variants_vep_all.csv')\n",
    "\n",
    "# keep dataframe in memory\n",
    "VEP_DF = pd.read_csv(VEP_OUTPUT_PATH, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>SWISSPROT</th>\n",
       "      <th>REF_ALLELE</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>CANONICAL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NC_000002.12:g.88978974G&gt;A</td>\n",
       "      <td>A0A075B6H7.48</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>4</td>\n",
       "      <td>P/L</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NC_000002.12:g.88978975G&gt;T</td>\n",
       "      <td>A0A075B6H7.48</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>4</td>\n",
       "      <td>P/T</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NC_000002.12:g.88978971G&gt;A</td>\n",
       "      <td>A0A075B6H7.48</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>5</td>\n",
       "      <td>A/V</td>\n",
       "      <td>-</td>\n",
       "      <td>2.033e-05</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NC_000002.12:g.88978972C&gt;T</td>\n",
       "      <td>A0A075B6H7.48</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>5</td>\n",
       "      <td>A/T</td>\n",
       "      <td>-</td>\n",
       "      <td>6.774e-06</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NC_000002.12:g.88978968T&gt;G</td>\n",
       "      <td>A0A075B6H7.48</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>6</td>\n",
       "      <td>Q/P</td>\n",
       "      <td>-</td>\n",
       "      <td>6.773e-06</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681953</th>\n",
       "      <td>NC_000001.11:g.158999996T&gt;C</td>\n",
       "      <td>W6CW81.33</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>106</td>\n",
       "      <td>H/R</td>\n",
       "      <td>-</td>\n",
       "      <td>2.63e-05</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681954</th>\n",
       "      <td>NC_000001.11:g.158999987G&gt;C</td>\n",
       "      <td>W6CW81.33</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>109</td>\n",
       "      <td>S/C</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681955</th>\n",
       "      <td>NC_000001.11:g.158999987G&gt;T</td>\n",
       "      <td>W6CW81.33</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>109</td>\n",
       "      <td>S/Y</td>\n",
       "      <td>-</td>\n",
       "      <td>1.314e-05</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681956</th>\n",
       "      <td>NC_000001.11:g.158999978G&gt;A</td>\n",
       "      <td>W6CW81.33</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>112</td>\n",
       "      <td>A/V</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681957</th>\n",
       "      <td>NC_000001.11:g.158999976A&gt;C</td>\n",
       "      <td>W6CW81.33</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>113</td>\n",
       "      <td>C/G</td>\n",
       "      <td>-</td>\n",
       "      <td>6.572e-06</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10681958 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  #Uploaded_variation      SWISSPROT REF_ALLELE Allele   \n",
       "0          NC_000002.12:g.88978974G>A  A0A075B6H7.48          G      A  \\\n",
       "1          NC_000002.12:g.88978975G>T  A0A075B6H7.48          G      T   \n",
       "2          NC_000002.12:g.88978971G>A  A0A075B6H7.48          G      A   \n",
       "3          NC_000002.12:g.88978972C>T  A0A075B6H7.48          C      T   \n",
       "4          NC_000002.12:g.88978968T>G  A0A075B6H7.48          T      G   \n",
       "...                               ...            ...        ...    ...   \n",
       "10681953  NC_000001.11:g.158999996T>C      W6CW81.33          T      C   \n",
       "10681954  NC_000001.11:g.158999987G>C      W6CW81.33          G      C   \n",
       "10681955  NC_000001.11:g.158999987G>T      W6CW81.33          G      T   \n",
       "10681956  NC_000001.11:g.158999978G>A      W6CW81.33          G      A   \n",
       "10681957  NC_000001.11:g.158999976A>C      W6CW81.33          A      C   \n",
       "\n",
       "               Consequence Protein_position Amino_acids CLIN_SIG gnomADg_AF   \n",
       "0         missense_variant                4         P/L        -          -  \\\n",
       "1         missense_variant                4         P/T        -          -   \n",
       "2         missense_variant                5         A/V        -  2.033e-05   \n",
       "3         missense_variant                5         A/T        -  6.774e-06   \n",
       "4         missense_variant                6         Q/P        -  6.773e-06   \n",
       "...                    ...              ...         ...      ...        ...   \n",
       "10681953  missense_variant              106         H/R        -   2.63e-05   \n",
       "10681954  missense_variant              109         S/C        -          -   \n",
       "10681955  missense_variant              109         S/Y        -  1.314e-05   \n",
       "10681956  missense_variant              112         A/V        -          -   \n",
       "10681957  missense_variant              113         C/G        -  6.572e-06   \n",
       "\n",
       "         CANONICAL  \n",
       "0              YES  \n",
       "1              YES  \n",
       "2              YES  \n",
       "3              YES  \n",
       "4              YES  \n",
       "...            ...  \n",
       "10681953       YES  \n",
       "10681954       YES  \n",
       "10681955       YES  \n",
       "10681956       YES  \n",
       "10681957       YES  \n",
       "\n",
       "[10681958 rows x 10 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "display(VEP_DF)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Try to get rows with benign/pathogenic disregarding ambiguity**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>SWISSPROT</th>\n",
       "      <th>REF_ALLELE</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>CANONICAL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>NC_000002.12:g.88978700A&gt;G</td>\n",
       "      <td>A0A075B6H7.48</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>39</td>\n",
       "      <td>V/A</td>\n",
       "      <td>-</td>\n",
       "      <td>0.0208</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>NC_000002.12:g.88978653T&gt;A</td>\n",
       "      <td>A0A075B6H7.48</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>55</td>\n",
       "      <td>T/S</td>\n",
       "      <td>-</td>\n",
       "      <td>0.0113</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>NC_000002.12:g.88978584T&gt;C</td>\n",
       "      <td>A0A075B6H7.48</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>78</td>\n",
       "      <td>S/G</td>\n",
       "      <td>-</td>\n",
       "      <td>0.2179</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>411</th>\n",
       "      <td>NC_000002.12:g.90190408C&gt;T</td>\n",
       "      <td>A0A075B6H8.46</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>27</td>\n",
       "      <td>T/I</td>\n",
       "      <td>-</td>\n",
       "      <td>0.7535</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>605</th>\n",
       "      <td>NC_000022.11:g.22201720G&gt;C</td>\n",
       "      <td>A0A075B6I3.37</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant,splice_region_variant</td>\n",
       "      <td>16</td>\n",
       "      <td>G/R</td>\n",
       "      <td>-</td>\n",
       "      <td>0.01368</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681651</th>\n",
       "      <td>NC_000008.11:g.119105892C&gt;T</td>\n",
       "      <td>Q9Y6Z7.155</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>179</td>\n",
       "      <td>R/W</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.001565</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681854</th>\n",
       "      <td>NC_000019.10:g.51704275G&gt;A</td>\n",
       "      <td>W5XKT8.43</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>246</td>\n",
       "      <td>G/S</td>\n",
       "      <td>-</td>\n",
       "      <td>0.02386</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681877</th>\n",
       "      <td>NC_000019.10:g.51704332T&gt;A</td>\n",
       "      <td>W5XKT8.43</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>265</td>\n",
       "      <td>W/R</td>\n",
       "      <td>-</td>\n",
       "      <td>0.01797</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681921</th>\n",
       "      <td>NC_000001.11:g.159000194A&gt;T</td>\n",
       "      <td>W6CW81.33</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>40</td>\n",
       "      <td>L/Q</td>\n",
       "      <td>-</td>\n",
       "      <td>0.01915</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681932</th>\n",
       "      <td>NC_000001.11:g.159000146T&gt;C</td>\n",
       "      <td>W6CW81.33</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>56</td>\n",
       "      <td>H/R</td>\n",
       "      <td>-</td>\n",
       "      <td>0.04404</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>222260 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  #Uploaded_variation      SWISSPROT REF_ALLELE Allele   \n",
       "59         NC_000002.12:g.88978700A>G  A0A075B6H7.48          A      G  \\\n",
       "95         NC_000002.12:g.88978653T>A  A0A075B6H7.48          T      A   \n",
       "143        NC_000002.12:g.88978584T>C  A0A075B6H7.48          T      C   \n",
       "411        NC_000002.12:g.90190408C>T  A0A075B6H8.46          C      T   \n",
       "605        NC_000022.11:g.22201720G>C  A0A075B6I3.37          G      C   \n",
       "...                               ...            ...        ...    ...   \n",
       "10681651  NC_000008.11:g.119105892C>T     Q9Y6Z7.155          C      T   \n",
       "10681854   NC_000019.10:g.51704275G>A      W5XKT8.43          G      A   \n",
       "10681877   NC_000019.10:g.51704332T>A      W5XKT8.43          T      A   \n",
       "10681921  NC_000001.11:g.159000194A>T      W6CW81.33          A      T   \n",
       "10681932  NC_000001.11:g.159000146T>C      W6CW81.33          T      C   \n",
       "\n",
       "                                     Consequence Protein_position Amino_acids   \n",
       "59                              missense_variant               39         V/A  \\\n",
       "95                              missense_variant               55         T/S   \n",
       "143                             missense_variant               78         S/G   \n",
       "411                             missense_variant               27         T/I   \n",
       "605       missense_variant,splice_region_variant               16         G/R   \n",
       "...                                          ...              ...         ...   \n",
       "10681651                        missense_variant              179         R/W   \n",
       "10681854                        missense_variant              246         G/S   \n",
       "10681877                        missense_variant              265         W/R   \n",
       "10681921                        missense_variant               40         L/Q   \n",
       "10681932                        missense_variant               56         H/R   \n",
       "\n",
       "               CLIN_SIG gnomADg_AF CANONICAL  \n",
       "59                    -     0.0208       YES  \n",
       "95                    -     0.0113       YES  \n",
       "143                   -     0.2179       YES  \n",
       "411                   -     0.7535       YES  \n",
       "605                   -    0.01368       YES  \n",
       "...                 ...        ...       ...  \n",
       "10681651  likely_benign   0.001565       YES  \n",
       "10681854              -    0.02386       YES  \n",
       "10681877              -    0.01797       YES  \n",
       "10681921              -    0.01915       YES  \n",
       "10681932              -    0.04404       YES  \n",
       "\n",
       "[222260 rows x 10 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>SWISSPROT</th>\n",
       "      <th>REF_ALLELE</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>CANONICAL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9703</th>\n",
       "      <td>NC_000014.9:g.23104829G&gt;A</td>\n",
       "      <td>A0A1B0GTW7.27</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>31</td>\n",
       "      <td>S/F</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>0.001065</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9922</th>\n",
       "      <td>NC_000014.9:g.23102250G&gt;A</td>\n",
       "      <td>A0A1B0GTW7.27</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>384</td>\n",
       "      <td>S/L</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>1.971e-05</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14558</th>\n",
       "      <td>NC_000007.14:g.139161018A&gt;G</td>\n",
       "      <td>A0AVF1.135</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>263</td>\n",
       "      <td>N/S</td>\n",
       "      <td>likely_pathogenic,pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14559</th>\n",
       "      <td>NC_000007.14:g.139161018A&gt;G</td>\n",
       "      <td>A0AVF1.135</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>263</td>\n",
       "      <td>N/S</td>\n",
       "      <td>likely_pathogenic,pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22222</th>\n",
       "      <td>NC_000007.14:g.122303281G&gt;A</td>\n",
       "      <td>A0PJY2.135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>278</td>\n",
       "      <td>H/Y</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10680087</th>\n",
       "      <td>NC_000001.11:g.7677682C&gt;T</td>\n",
       "      <td>Q9Y6Y1.171</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>955</td>\n",
       "      <td>R/W</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10680187</th>\n",
       "      <td>NC_000001.11:g.7736507A&gt;G</td>\n",
       "      <td>Q9Y6Y1.171</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1077</td>\n",
       "      <td>Y/C</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10680188</th>\n",
       "      <td>NC_000001.11:g.7736507A&gt;G</td>\n",
       "      <td>Q9Y6Y1.171</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1077</td>\n",
       "      <td>Y/C</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681645</th>\n",
       "      <td>NC_000008.11:g.119105885C&gt;G</td>\n",
       "      <td>Q9Y6Z7.155</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>176</td>\n",
       "      <td>C/W</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681647</th>\n",
       "      <td>NC_000008.11:g.119105885C&gt;G</td>\n",
       "      <td>Q9Y6Z7.155</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>176</td>\n",
       "      <td>C/W</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>207435 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  #Uploaded_variation      SWISSPROT REF_ALLELE Allele   \n",
       "9703        NC_000014.9:g.23104829G>A  A0A1B0GTW7.27          G      A  \\\n",
       "9922        NC_000014.9:g.23102250G>A  A0A1B0GTW7.27          G      A   \n",
       "14558     NC_000007.14:g.139161018A>G     A0AVF1.135          A      G   \n",
       "14559     NC_000007.14:g.139161018A>G     A0AVF1.135          A      G   \n",
       "22222     NC_000007.14:g.122303281G>A     A0PJY2.135          G      A   \n",
       "...                               ...            ...        ...    ...   \n",
       "10680087    NC_000001.11:g.7677682C>T     Q9Y6Y1.171          C      T   \n",
       "10680187    NC_000001.11:g.7736507A>G     Q9Y6Y1.171          A      G   \n",
       "10680188    NC_000001.11:g.7736507A>G     Q9Y6Y1.171          A      G   \n",
       "10681645  NC_000008.11:g.119105885C>G     Q9Y6Z7.155          C      G   \n",
       "10681647  NC_000008.11:g.119105885C>G     Q9Y6Z7.155          C      G   \n",
       "\n",
       "               Consequence Protein_position Amino_acids   \n",
       "9703      missense_variant               31         S/F  \\\n",
       "9922      missense_variant              384         S/L   \n",
       "14558     missense_variant              263         N/S   \n",
       "14559     missense_variant              263         N/S   \n",
       "22222     missense_variant              278         H/Y   \n",
       "...                    ...              ...         ...   \n",
       "10680087  missense_variant              955         R/W   \n",
       "10680187  missense_variant             1077         Y/C   \n",
       "10680188  missense_variant             1077         Y/C   \n",
       "10681645  missense_variant              176         C/W   \n",
       "10681647  missense_variant              176         C/W   \n",
       "\n",
       "                              CLIN_SIG gnomADg_AF CANONICAL  \n",
       "9703                        pathogenic   0.001065       YES  \n",
       "9922                        pathogenic  1.971e-05       YES  \n",
       "14558     likely_pathogenic,pathogenic          -       YES  \n",
       "14559     likely_pathogenic,pathogenic          -       YES  \n",
       "22222                       pathogenic          -       YES  \n",
       "...                                ...        ...       ...  \n",
       "10680087             likely_pathogenic          -       YES  \n",
       "10680187             likely_pathogenic          -       YES  \n",
       "10680188             likely_pathogenic          -       YES  \n",
       "10681645                    pathogenic          -       YES  \n",
       "10681647                    pathogenic          -       YES  \n",
       "\n",
       "[207435 rows x 10 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "import re\n",
    "\n",
    "benignRegex = r'(benign)'\n",
    "justBenignRegex = r'(benign)'\n",
    "pathogenicRegex = r'(pathogenic|damaging)'\n",
    "\n",
    "def evaluate_AF(AF):\n",
    "   \n",
    "   try:\n",
    "      flt = float(AF.values)\n",
    "      return flt\n",
    "   except TypeError:\n",
    "      return -1\n",
    "   \n",
    "def convert_AF(AF):\n",
    "   \n",
    "   try:\n",
    "      return float(AF)\n",
    "   except ValueError:\n",
    "      return -1\n",
    "\n",
    "# benign\n",
    "# vep_all_benign_df = VEP_NR[VEP_NR['CLIN_SIG'].str.contains(benignRegex, regex=True) & VEP_NR['Consequence'].str.contains(consequenceRegex, regex=True)]\n",
    "# benign_high_AF = vep_all_benign_df['gnomADg_AF'].apply(convert_AF) >= 0.01\n",
    "# vep_all_benign_df = vep_all_benign_df[VEP_NR['CLIN_SIG'].str.contains(justBenignRegex, regex=True) | benign_high_AF]\n",
    "\n",
    "benignMask = []\n",
    "\n",
    "def to_float(numStr):\n",
    "   \n",
    "   try:\n",
    "      return float(numStr)\n",
    "   except ValueError:\n",
    "      return -1\n",
    "   \n",
    "for row in VEP_DF.itertuples():\n",
    "   \n",
    "   CLIN_SIG = getattr(row, 'CLIN_SIG')\n",
    "   AF = to_float(getattr(row, 'gnomADg_AF'))\n",
    "   consequence = getattr(row, 'Consequence')\n",
    "   \n",
    "   isBenign = bool(re.search(benignRegex, CLIN_SIG))\n",
    "   isPathogenic = bool(re.search(pathogenicRegex, CLIN_SIG))\n",
    "   \n",
    "   if (isBenign or AF >= 0.01) and isPathogenic == False:\n",
    "      benignMask.append(True)\n",
    "   else:\n",
    "      benignMask.append(False)\n",
    "\n",
    "vep_all_benign_df = VEP_DF[benignMask]\n",
    "\n",
    "# pathogenic\n",
    "pathogenicMask = []\n",
    "\n",
    "for row in VEP_DF.itertuples():\n",
    "   \n",
    "   CLIN_SIG = getattr(row, 'CLIN_SIG')\n",
    "   consequence = getattr(row, 'Consequence')\n",
    "   \n",
    "   isPathogenic = bool(re.search(pathogenicRegex, CLIN_SIG))\n",
    "   \n",
    "   if isPathogenic:\n",
    "      pathogenicMask.append(True)\n",
    "   else:\n",
    "      pathogenicMask.append(False)\n",
    "\n",
    "vep_all_pathogenic_df = VEP_DF[pathogenicMask]\n",
    "\n",
    "display(vep_all_benign_df)\n",
    "display(vep_all_pathogenic_df)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Check how many rows reduced when you exclude ambiguous mutations**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/y3/mxnsyg_910x11q_t3z54wkbr0000gn/T/ipykernel_39382/1978656979.py:5: UserWarning: This pattern is interpreted as a regular expression, and has match groups. To actually get the groups, use str.extract.\n",
      "  vep_unamb_benign_df = vep_all_benign_df[~vep_all_benign_df['CLIN_SIG'].str.contains(ambiguousRegex, regex=True)]\n",
      "/var/folders/y3/mxnsyg_910x11q_t3z54wkbr0000gn/T/ipykernel_39382/1978656979.py:6: UserWarning: This pattern is interpreted as a regular expression, and has match groups. To actually get the groups, use str.extract.\n",
      "  vep_unamb_pathogenic_df = vep_all_pathogenic_df[~vep_all_pathogenic_df['CLIN_SIG'].str.contains(ambiguousRegex, regex=True)]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>SWISSPROT</th>\n",
       "      <th>REF_ALLELE</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>CANONICAL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>NC_000002.12:g.88978700A&gt;G</td>\n",
       "      <td>A0A075B6H7.48</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>39</td>\n",
       "      <td>V/A</td>\n",
       "      <td>-</td>\n",
       "      <td>0.0208</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>NC_000002.12:g.88978653T&gt;A</td>\n",
       "      <td>A0A075B6H7.48</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>55</td>\n",
       "      <td>T/S</td>\n",
       "      <td>-</td>\n",
       "      <td>0.0113</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>NC_000002.12:g.88978584T&gt;C</td>\n",
       "      <td>A0A075B6H7.48</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>78</td>\n",
       "      <td>S/G</td>\n",
       "      <td>-</td>\n",
       "      <td>0.2179</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>411</th>\n",
       "      <td>NC_000002.12:g.90190408C&gt;T</td>\n",
       "      <td>A0A075B6H8.46</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>27</td>\n",
       "      <td>T/I</td>\n",
       "      <td>-</td>\n",
       "      <td>0.7535</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>605</th>\n",
       "      <td>NC_000022.11:g.22201720G&gt;C</td>\n",
       "      <td>A0A075B6I3.37</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant,splice_region_variant</td>\n",
       "      <td>16</td>\n",
       "      <td>G/R</td>\n",
       "      <td>-</td>\n",
       "      <td>0.01368</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681651</th>\n",
       "      <td>NC_000008.11:g.119105892C&gt;T</td>\n",
       "      <td>Q9Y6Z7.155</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>179</td>\n",
       "      <td>R/W</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.001565</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681854</th>\n",
       "      <td>NC_000019.10:g.51704275G&gt;A</td>\n",
       "      <td>W5XKT8.43</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>246</td>\n",
       "      <td>G/S</td>\n",
       "      <td>-</td>\n",
       "      <td>0.02386</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681877</th>\n",
       "      <td>NC_000019.10:g.51704332T&gt;A</td>\n",
       "      <td>W5XKT8.43</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>265</td>\n",
       "      <td>W/R</td>\n",
       "      <td>-</td>\n",
       "      <td>0.01797</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681921</th>\n",
       "      <td>NC_000001.11:g.159000194A&gt;T</td>\n",
       "      <td>W6CW81.33</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>40</td>\n",
       "      <td>L/Q</td>\n",
       "      <td>-</td>\n",
       "      <td>0.01915</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681932</th>\n",
       "      <td>NC_000001.11:g.159000146T&gt;C</td>\n",
       "      <td>W6CW81.33</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>56</td>\n",
       "      <td>H/R</td>\n",
       "      <td>-</td>\n",
       "      <td>0.04404</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>190286 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  #Uploaded_variation      SWISSPROT REF_ALLELE Allele   \n",
       "59         NC_000002.12:g.88978700A>G  A0A075B6H7.48          A      G  \\\n",
       "95         NC_000002.12:g.88978653T>A  A0A075B6H7.48          T      A   \n",
       "143        NC_000002.12:g.88978584T>C  A0A075B6H7.48          T      C   \n",
       "411        NC_000002.12:g.90190408C>T  A0A075B6H8.46          C      T   \n",
       "605        NC_000022.11:g.22201720G>C  A0A075B6I3.37          G      C   \n",
       "...                               ...            ...        ...    ...   \n",
       "10681651  NC_000008.11:g.119105892C>T     Q9Y6Z7.155          C      T   \n",
       "10681854   NC_000019.10:g.51704275G>A      W5XKT8.43          G      A   \n",
       "10681877   NC_000019.10:g.51704332T>A      W5XKT8.43          T      A   \n",
       "10681921  NC_000001.11:g.159000194A>T      W6CW81.33          A      T   \n",
       "10681932  NC_000001.11:g.159000146T>C      W6CW81.33          T      C   \n",
       "\n",
       "                                     Consequence Protein_position Amino_acids   \n",
       "59                              missense_variant               39         V/A  \\\n",
       "95                              missense_variant               55         T/S   \n",
       "143                             missense_variant               78         S/G   \n",
       "411                             missense_variant               27         T/I   \n",
       "605       missense_variant,splice_region_variant               16         G/R   \n",
       "...                                          ...              ...         ...   \n",
       "10681651                        missense_variant              179         R/W   \n",
       "10681854                        missense_variant              246         G/S   \n",
       "10681877                        missense_variant              265         W/R   \n",
       "10681921                        missense_variant               40         L/Q   \n",
       "10681932                        missense_variant               56         H/R   \n",
       "\n",
       "               CLIN_SIG gnomADg_AF CANONICAL  \n",
       "59                    -     0.0208       YES  \n",
       "95                    -     0.0113       YES  \n",
       "143                   -     0.2179       YES  \n",
       "411                   -     0.7535       YES  \n",
       "605                   -    0.01368       YES  \n",
       "...                 ...        ...       ...  \n",
       "10681651  likely_benign   0.001565       YES  \n",
       "10681854              -    0.02386       YES  \n",
       "10681877              -    0.01797       YES  \n",
       "10681921              -    0.01915       YES  \n",
       "10681932              -    0.04404       YES  \n",
       "\n",
       "[190286 rows x 10 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>SWISSPROT</th>\n",
       "      <th>REF_ALLELE</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>CANONICAL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9703</th>\n",
       "      <td>NC_000014.9:g.23104829G&gt;A</td>\n",
       "      <td>A0A1B0GTW7.27</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>31</td>\n",
       "      <td>S/F</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>0.001065</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9922</th>\n",
       "      <td>NC_000014.9:g.23102250G&gt;A</td>\n",
       "      <td>A0A1B0GTW7.27</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>384</td>\n",
       "      <td>S/L</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>1.971e-05</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14558</th>\n",
       "      <td>NC_000007.14:g.139161018A&gt;G</td>\n",
       "      <td>A0AVF1.135</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>263</td>\n",
       "      <td>N/S</td>\n",
       "      <td>likely_pathogenic,pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14559</th>\n",
       "      <td>NC_000007.14:g.139161018A&gt;G</td>\n",
       "      <td>A0AVF1.135</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>263</td>\n",
       "      <td>N/S</td>\n",
       "      <td>likely_pathogenic,pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22222</th>\n",
       "      <td>NC_000007.14:g.122303281G&gt;A</td>\n",
       "      <td>A0PJY2.135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>278</td>\n",
       "      <td>H/Y</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10680087</th>\n",
       "      <td>NC_000001.11:g.7677682C&gt;T</td>\n",
       "      <td>Q9Y6Y1.171</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>955</td>\n",
       "      <td>R/W</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10680187</th>\n",
       "      <td>NC_000001.11:g.7736507A&gt;G</td>\n",
       "      <td>Q9Y6Y1.171</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1077</td>\n",
       "      <td>Y/C</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10680188</th>\n",
       "      <td>NC_000001.11:g.7736507A&gt;G</td>\n",
       "      <td>Q9Y6Y1.171</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1077</td>\n",
       "      <td>Y/C</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681645</th>\n",
       "      <td>NC_000008.11:g.119105885C&gt;G</td>\n",
       "      <td>Q9Y6Z7.155</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>176</td>\n",
       "      <td>C/W</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681647</th>\n",
       "      <td>NC_000008.11:g.119105885C&gt;G</td>\n",
       "      <td>Q9Y6Z7.155</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>176</td>\n",
       "      <td>C/W</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>152650 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  #Uploaded_variation      SWISSPROT REF_ALLELE Allele   \n",
       "9703        NC_000014.9:g.23104829G>A  A0A1B0GTW7.27          G      A  \\\n",
       "9922        NC_000014.9:g.23102250G>A  A0A1B0GTW7.27          G      A   \n",
       "14558     NC_000007.14:g.139161018A>G     A0AVF1.135          A      G   \n",
       "14559     NC_000007.14:g.139161018A>G     A0AVF1.135          A      G   \n",
       "22222     NC_000007.14:g.122303281G>A     A0PJY2.135          G      A   \n",
       "...                               ...            ...        ...    ...   \n",
       "10680087    NC_000001.11:g.7677682C>T     Q9Y6Y1.171          C      T   \n",
       "10680187    NC_000001.11:g.7736507A>G     Q9Y6Y1.171          A      G   \n",
       "10680188    NC_000001.11:g.7736507A>G     Q9Y6Y1.171          A      G   \n",
       "10681645  NC_000008.11:g.119105885C>G     Q9Y6Z7.155          C      G   \n",
       "10681647  NC_000008.11:g.119105885C>G     Q9Y6Z7.155          C      G   \n",
       "\n",
       "               Consequence Protein_position Amino_acids   \n",
       "9703      missense_variant               31         S/F  \\\n",
       "9922      missense_variant              384         S/L   \n",
       "14558     missense_variant              263         N/S   \n",
       "14559     missense_variant              263         N/S   \n",
       "22222     missense_variant              278         H/Y   \n",
       "...                    ...              ...         ...   \n",
       "10680087  missense_variant              955         R/W   \n",
       "10680187  missense_variant             1077         Y/C   \n",
       "10680188  missense_variant             1077         Y/C   \n",
       "10681645  missense_variant              176         C/W   \n",
       "10681647  missense_variant              176         C/W   \n",
       "\n",
       "                              CLIN_SIG gnomADg_AF CANONICAL  \n",
       "9703                        pathogenic   0.001065       YES  \n",
       "9922                        pathogenic  1.971e-05       YES  \n",
       "14558     likely_pathogenic,pathogenic          -       YES  \n",
       "14559     likely_pathogenic,pathogenic          -       YES  \n",
       "22222                       pathogenic          -       YES  \n",
       "...                                ...        ...       ...  \n",
       "10680087             likely_pathogenic          -       YES  \n",
       "10680187             likely_pathogenic          -       YES  \n",
       "10680188             likely_pathogenic          -       YES  \n",
       "10681645                    pathogenic          -       YES  \n",
       "10681647                    pathogenic          -       YES  \n",
       "\n",
       "[152650 rows x 10 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "ambiguousRegex = r'(uncertain|conflicting)'\n",
    "\n",
    "vep_unamb_benign_df = vep_all_benign_df[~vep_all_benign_df['CLIN_SIG'].str.contains(ambiguousRegex, regex=True)]\n",
    "vep_unamb_pathogenic_df = vep_all_pathogenic_df[~vep_all_pathogenic_df['CLIN_SIG'].str.contains(ambiguousRegex, regex=True)]\n",
    "\n",
    "VEP_BENIGN_OUTPUT = os.path.join(TROUBLESHOOT_PATH, 'vep_output_benign_variants.csv')\n",
    "VEP_PATHOGENIC_OUTPUT = os.path.join(TROUBLESHOOT_PATH, 'vep_output_pathogenic_variants.csv')\n",
    "\n",
    "vep_unamb_benign_df.to_csv(VEP_BENIGN_OUTPUT, sep='\\t', index=False)\n",
    "vep_unamb_pathogenic_df.to_csv(VEP_PATHOGENIC_OUTPUT, sep='\\t', index=False)\n",
    "\n",
    "display(vep_unamb_benign_df)\n",
    "display(vep_unamb_pathogenic_df)\n",
    "\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Exclude Non-reviewed Proteins**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>SWISSPROT</th>\n",
       "      <th>REF_ALLELE</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>CANONICAL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>NC_000002.12:g.88978700A&gt;G</td>\n",
       "      <td>A0A075B6H7</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>39</td>\n",
       "      <td>V/A</td>\n",
       "      <td>-</td>\n",
       "      <td>0.0208</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>NC_000002.12:g.88978653T&gt;A</td>\n",
       "      <td>A0A075B6H7</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>55</td>\n",
       "      <td>T/S</td>\n",
       "      <td>-</td>\n",
       "      <td>0.0113</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>NC_000002.12:g.88978584T&gt;C</td>\n",
       "      <td>A0A075B6H7</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>78</td>\n",
       "      <td>S/G</td>\n",
       "      <td>-</td>\n",
       "      <td>0.2179</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>411</th>\n",
       "      <td>NC_000002.12:g.90190408C&gt;T</td>\n",
       "      <td>A0A075B6H8</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>27</td>\n",
       "      <td>T/I</td>\n",
       "      <td>-</td>\n",
       "      <td>0.7535</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>605</th>\n",
       "      <td>NC_000022.11:g.22201720G&gt;C</td>\n",
       "      <td>A0A075B6I3</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant,splice_region_variant</td>\n",
       "      <td>16</td>\n",
       "      <td>G/R</td>\n",
       "      <td>-</td>\n",
       "      <td>0.01368</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681651</th>\n",
       "      <td>NC_000008.11:g.119105892C&gt;T</td>\n",
       "      <td>Q9Y6Z7</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>179</td>\n",
       "      <td>R/W</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.001565</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681854</th>\n",
       "      <td>NC_000019.10:g.51704275G&gt;A</td>\n",
       "      <td>W5XKT8</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>246</td>\n",
       "      <td>G/S</td>\n",
       "      <td>-</td>\n",
       "      <td>0.02386</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681877</th>\n",
       "      <td>NC_000019.10:g.51704332T&gt;A</td>\n",
       "      <td>W5XKT8</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>265</td>\n",
       "      <td>W/R</td>\n",
       "      <td>-</td>\n",
       "      <td>0.01797</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681921</th>\n",
       "      <td>NC_000001.11:g.159000194A&gt;T</td>\n",
       "      <td>W6CW81</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>40</td>\n",
       "      <td>L/Q</td>\n",
       "      <td>-</td>\n",
       "      <td>0.01915</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681932</th>\n",
       "      <td>NC_000001.11:g.159000146T&gt;C</td>\n",
       "      <td>W6CW81</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>56</td>\n",
       "      <td>H/R</td>\n",
       "      <td>-</td>\n",
       "      <td>0.04404</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>184523 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  #Uploaded_variation   SWISSPROT REF_ALLELE Allele   \n",
       "59         NC_000002.12:g.88978700A>G  A0A075B6H7          A      G  \\\n",
       "95         NC_000002.12:g.88978653T>A  A0A075B6H7          T      A   \n",
       "143        NC_000002.12:g.88978584T>C  A0A075B6H7          T      C   \n",
       "411        NC_000002.12:g.90190408C>T  A0A075B6H8          C      T   \n",
       "605        NC_000022.11:g.22201720G>C  A0A075B6I3          G      C   \n",
       "...                               ...         ...        ...    ...   \n",
       "10681651  NC_000008.11:g.119105892C>T      Q9Y6Z7          C      T   \n",
       "10681854   NC_000019.10:g.51704275G>A      W5XKT8          G      A   \n",
       "10681877   NC_000019.10:g.51704332T>A      W5XKT8          T      A   \n",
       "10681921  NC_000001.11:g.159000194A>T      W6CW81          A      T   \n",
       "10681932  NC_000001.11:g.159000146T>C      W6CW81          T      C   \n",
       "\n",
       "                                     Consequence Protein_position Amino_acids   \n",
       "59                              missense_variant               39         V/A  \\\n",
       "95                              missense_variant               55         T/S   \n",
       "143                             missense_variant               78         S/G   \n",
       "411                             missense_variant               27         T/I   \n",
       "605       missense_variant,splice_region_variant               16         G/R   \n",
       "...                                          ...              ...         ...   \n",
       "10681651                        missense_variant              179         R/W   \n",
       "10681854                        missense_variant              246         G/S   \n",
       "10681877                        missense_variant              265         W/R   \n",
       "10681921                        missense_variant               40         L/Q   \n",
       "10681932                        missense_variant               56         H/R   \n",
       "\n",
       "               CLIN_SIG gnomADg_AF CANONICAL  \n",
       "59                    -     0.0208       YES  \n",
       "95                    -     0.0113       YES  \n",
       "143                   -     0.2179       YES  \n",
       "411                   -     0.7535       YES  \n",
       "605                   -    0.01368       YES  \n",
       "...                 ...        ...       ...  \n",
       "10681651  likely_benign   0.001565       YES  \n",
       "10681854              -    0.02386       YES  \n",
       "10681877              -    0.01797       YES  \n",
       "10681921              -    0.01915       YES  \n",
       "10681932              -    0.04404       YES  \n",
       "\n",
       "[184523 rows x 10 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>SWISSPROT</th>\n",
       "      <th>REF_ALLELE</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>CANONICAL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9703</th>\n",
       "      <td>NC_000014.9:g.23104829G&gt;A</td>\n",
       "      <td>A0A1B0GTW7</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>31</td>\n",
       "      <td>S/F</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>0.001065</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9922</th>\n",
       "      <td>NC_000014.9:g.23102250G&gt;A</td>\n",
       "      <td>A0A1B0GTW7</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>384</td>\n",
       "      <td>S/L</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>1.971e-05</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14558</th>\n",
       "      <td>NC_000007.14:g.139161018A&gt;G</td>\n",
       "      <td>A0AVF1</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>263</td>\n",
       "      <td>N/S</td>\n",
       "      <td>likely_pathogenic,pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14559</th>\n",
       "      <td>NC_000007.14:g.139161018A&gt;G</td>\n",
       "      <td>A0AVF1</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>263</td>\n",
       "      <td>N/S</td>\n",
       "      <td>likely_pathogenic,pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22222</th>\n",
       "      <td>NC_000007.14:g.122303281G&gt;A</td>\n",
       "      <td>A0PJY2</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>278</td>\n",
       "      <td>H/Y</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10680087</th>\n",
       "      <td>NC_000001.11:g.7677682C&gt;T</td>\n",
       "      <td>Q9Y6Y1</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>955</td>\n",
       "      <td>R/W</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10680187</th>\n",
       "      <td>NC_000001.11:g.7736507A&gt;G</td>\n",
       "      <td>Q9Y6Y1</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1077</td>\n",
       "      <td>Y/C</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10680188</th>\n",
       "      <td>NC_000001.11:g.7736507A&gt;G</td>\n",
       "      <td>Q9Y6Y1</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1077</td>\n",
       "      <td>Y/C</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681645</th>\n",
       "      <td>NC_000008.11:g.119105885C&gt;G</td>\n",
       "      <td>Q9Y6Z7</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>176</td>\n",
       "      <td>C/W</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681647</th>\n",
       "      <td>NC_000008.11:g.119105885C&gt;G</td>\n",
       "      <td>Q9Y6Z7</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>176</td>\n",
       "      <td>C/W</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>150738 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  #Uploaded_variation   SWISSPROT REF_ALLELE Allele   \n",
       "9703        NC_000014.9:g.23104829G>A  A0A1B0GTW7          G      A  \\\n",
       "9922        NC_000014.9:g.23102250G>A  A0A1B0GTW7          G      A   \n",
       "14558     NC_000007.14:g.139161018A>G      A0AVF1          A      G   \n",
       "14559     NC_000007.14:g.139161018A>G      A0AVF1          A      G   \n",
       "22222     NC_000007.14:g.122303281G>A      A0PJY2          G      A   \n",
       "...                               ...         ...        ...    ...   \n",
       "10680087    NC_000001.11:g.7677682C>T      Q9Y6Y1          C      T   \n",
       "10680187    NC_000001.11:g.7736507A>G      Q9Y6Y1          A      G   \n",
       "10680188    NC_000001.11:g.7736507A>G      Q9Y6Y1          A      G   \n",
       "10681645  NC_000008.11:g.119105885C>G      Q9Y6Z7          C      G   \n",
       "10681647  NC_000008.11:g.119105885C>G      Q9Y6Z7          C      G   \n",
       "\n",
       "               Consequence Protein_position Amino_acids   \n",
       "9703      missense_variant               31         S/F  \\\n",
       "9922      missense_variant              384         S/L   \n",
       "14558     missense_variant              263         N/S   \n",
       "14559     missense_variant              263         N/S   \n",
       "22222     missense_variant              278         H/Y   \n",
       "...                    ...              ...         ...   \n",
       "10680087  missense_variant              955         R/W   \n",
       "10680187  missense_variant             1077         Y/C   \n",
       "10680188  missense_variant             1077         Y/C   \n",
       "10681645  missense_variant              176         C/W   \n",
       "10681647  missense_variant              176         C/W   \n",
       "\n",
       "                              CLIN_SIG gnomADg_AF CANONICAL  \n",
       "9703                        pathogenic   0.001065       YES  \n",
       "9922                        pathogenic  1.971e-05       YES  \n",
       "14558     likely_pathogenic,pathogenic          -       YES  \n",
       "14559     likely_pathogenic,pathogenic          -       YES  \n",
       "22222                       pathogenic          -       YES  \n",
       "...                                ...        ...       ...  \n",
       "10680087             likely_pathogenic          -       YES  \n",
       "10680187             likely_pathogenic          -       YES  \n",
       "10680188             likely_pathogenic          -       YES  \n",
       "10681645                    pathogenic          -       YES  \n",
       "10681647                    pathogenic          -       YES  \n",
       "\n",
       "[150738 rows x 10 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "REVIEWED_UNIPROT_PATH = os.path.join(ROOT_DIR, 'files/uniprot_human_reviewed_lt_2700aa.csv')\n",
    "from utils import readFile_as_generator\n",
    "\n",
    "REVIEWED_UNIPROT = {}\n",
    "\n",
    "def std_uniprot(uniprot: str):\n",
    "\n",
    "   pattern = r'\\..*'\n",
    "   \n",
    "   return uniprot.replace(pattern, '', regex=True)\n",
    "\n",
    "def check_protein(uniprot):\n",
    "   try:\n",
    "      value = REVIEWED_UNIPROT[str(uniprot)]\n",
    "      return value\n",
    "   except KeyError:\n",
    "      return False\n",
    "   \n",
    "for protein in readFile_as_generator(REVIEWED_UNIPROT_PATH):\n",
    "   REVIEWED_UNIPROT[protein.strip('\\n')] = True\n",
    "\n",
    "# print(REVIEWED_UNIPROT)\n",
    "\n",
    "# standardize SWISSPROT column\n",
    "vep_unamb_benign_df['SWISSPROT'] = std_uniprot(vep_unamb_benign_df['SWISSPROT'])\n",
    "vep_unamb_pathogenic_df['SWISSPROT'] = std_uniprot(vep_unamb_pathogenic_df['SWISSPROT'])\n",
    "\n",
    "reviewed_benign_rows = vep_unamb_benign_df['SWISSPROT'].apply(check_protein)\n",
    "reviewed_pathogenic_rows = vep_unamb_pathogenic_df['SWISSPROT'].apply(check_protein)\n",
    "\n",
    "vep_benign_reviewed_df = vep_unamb_benign_df[reviewed_benign_rows]\n",
    "vep_pathogenic_reviewed_df = vep_unamb_pathogenic_df[reviewed_pathogenic_rows]\n",
    "\n",
    "FILES_PATH = os.path.join(ROOT_DIR, 'files/vep_benign_redundant.csv')\n",
    "\n",
    "vep_benign_reviewed_df.to_csv(FILES_PATH, sep='\\t', index=False)\n",
    "\n",
    "display(vep_benign_reviewed_df)\n",
    "display(vep_pathogenic_reviewed_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>SWISSPROT</th>\n",
       "      <th>REF_ALLELE</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>CANONICAL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>NC_000002.12:g.88978700A&gt;G</td>\n",
       "      <td>A0A075B6H7</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>39</td>\n",
       "      <td>V/A</td>\n",
       "      <td>-</td>\n",
       "      <td>0.0208</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>NC_000002.12:g.88978653T&gt;A</td>\n",
       "      <td>A0A075B6H7</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>55</td>\n",
       "      <td>T/S</td>\n",
       "      <td>-</td>\n",
       "      <td>0.0113</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>NC_000002.12:g.88978584T&gt;C</td>\n",
       "      <td>A0A075B6H7</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>78</td>\n",
       "      <td>S/G</td>\n",
       "      <td>-</td>\n",
       "      <td>0.2179</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>411</th>\n",
       "      <td>NC_000002.12:g.90190408C&gt;T</td>\n",
       "      <td>A0A075B6H8</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>27</td>\n",
       "      <td>T/I</td>\n",
       "      <td>-</td>\n",
       "      <td>0.7535</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>647</th>\n",
       "      <td>NC_000022.11:g.22201974T&gt;C</td>\n",
       "      <td>A0A075B6I3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>61</td>\n",
       "      <td>L/P</td>\n",
       "      <td>-</td>\n",
       "      <td>0.4648</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681650</th>\n",
       "      <td>NC_000008.11:g.119105892C&gt;T</td>\n",
       "      <td>Q9Y6Z7</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>179</td>\n",
       "      <td>R/W</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.001565</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681854</th>\n",
       "      <td>NC_000019.10:g.51704275G&gt;A</td>\n",
       "      <td>W5XKT8</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>246</td>\n",
       "      <td>G/S</td>\n",
       "      <td>-</td>\n",
       "      <td>0.02386</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681877</th>\n",
       "      <td>NC_000019.10:g.51704332T&gt;A</td>\n",
       "      <td>W5XKT8</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>265</td>\n",
       "      <td>W/R</td>\n",
       "      <td>-</td>\n",
       "      <td>0.01797</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681921</th>\n",
       "      <td>NC_000001.11:g.159000194A&gt;T</td>\n",
       "      <td>W6CW81</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>40</td>\n",
       "      <td>L/Q</td>\n",
       "      <td>-</td>\n",
       "      <td>0.01915</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681932</th>\n",
       "      <td>NC_000001.11:g.159000146T&gt;C</td>\n",
       "      <td>W6CW81</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>56</td>\n",
       "      <td>H/R</td>\n",
       "      <td>-</td>\n",
       "      <td>0.04404</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>58246 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  #Uploaded_variation   SWISSPROT REF_ALLELE Allele   \n",
       "59         NC_000002.12:g.88978700A>G  A0A075B6H7          A      G  \\\n",
       "95         NC_000002.12:g.88978653T>A  A0A075B6H7          T      A   \n",
       "143        NC_000002.12:g.88978584T>C  A0A075B6H7          T      C   \n",
       "411        NC_000002.12:g.90190408C>T  A0A075B6H8          C      T   \n",
       "647        NC_000022.11:g.22201974T>C  A0A075B6I3          T      C   \n",
       "...                               ...         ...        ...    ...   \n",
       "10681650  NC_000008.11:g.119105892C>T      Q9Y6Z7          C      T   \n",
       "10681854   NC_000019.10:g.51704275G>A      W5XKT8          G      A   \n",
       "10681877   NC_000019.10:g.51704332T>A      W5XKT8          T      A   \n",
       "10681921  NC_000001.11:g.159000194A>T      W6CW81          A      T   \n",
       "10681932  NC_000001.11:g.159000146T>C      W6CW81          T      C   \n",
       "\n",
       "               Consequence Protein_position Amino_acids       CLIN_SIG   \n",
       "59        missense_variant               39         V/A              -  \\\n",
       "95        missense_variant               55         T/S              -   \n",
       "143       missense_variant               78         S/G              -   \n",
       "411       missense_variant               27         T/I              -   \n",
       "647       missense_variant               61         L/P              -   \n",
       "...                    ...              ...         ...            ...   \n",
       "10681650  missense_variant              179         R/W  likely_benign   \n",
       "10681854  missense_variant              246         G/S              -   \n",
       "10681877  missense_variant              265         W/R              -   \n",
       "10681921  missense_variant               40         L/Q              -   \n",
       "10681932  missense_variant               56         H/R              -   \n",
       "\n",
       "         gnomADg_AF CANONICAL  \n",
       "59           0.0208       YES  \n",
       "95           0.0113       YES  \n",
       "143          0.2179       YES  \n",
       "411          0.7535       YES  \n",
       "647          0.4648       YES  \n",
       "...             ...       ...  \n",
       "10681650   0.001565       YES  \n",
       "10681854    0.02386       YES  \n",
       "10681877    0.01797       YES  \n",
       "10681921    0.01915       YES  \n",
       "10681932    0.04404       YES  \n",
       "\n",
       "[58246 rows x 10 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>SWISSPROT</th>\n",
       "      <th>REF_ALLELE</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>CANONICAL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9703</th>\n",
       "      <td>NC_000014.9:g.23104829G&gt;A</td>\n",
       "      <td>A0A1B0GTW7</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>31</td>\n",
       "      <td>S/F</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>0.001065</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9922</th>\n",
       "      <td>NC_000014.9:g.23102250G&gt;A</td>\n",
       "      <td>A0A1B0GTW7</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>384</td>\n",
       "      <td>S/L</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>1.971e-05</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14558</th>\n",
       "      <td>NC_000007.14:g.139161018A&gt;G</td>\n",
       "      <td>A0AVF1</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>263</td>\n",
       "      <td>N/S</td>\n",
       "      <td>likely_pathogenic,pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22222</th>\n",
       "      <td>NC_000007.14:g.122303281G&gt;A</td>\n",
       "      <td>A0PJY2</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>278</td>\n",
       "      <td>H/Y</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23235</th>\n",
       "      <td>NC_000004.12:g.17526877C&gt;A</td>\n",
       "      <td>A0PK11</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>165</td>\n",
       "      <td>T/K</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10678657</th>\n",
       "      <td>NC_000022.11:g.30937947C&gt;A</td>\n",
       "      <td>Q9Y6X9</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>413</td>\n",
       "      <td>V/F</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10678661</th>\n",
       "      <td>NC_000022.11:g.30937892G&gt;A</td>\n",
       "      <td>Q9Y6X9</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>431</td>\n",
       "      <td>A/V</td>\n",
       "      <td>pathogenic,likely_pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10680086</th>\n",
       "      <td>NC_000001.11:g.7677682C&gt;T</td>\n",
       "      <td>Q9Y6Y1</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>955</td>\n",
       "      <td>R/W</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10680187</th>\n",
       "      <td>NC_000001.11:g.7736507A&gt;G</td>\n",
       "      <td>Q9Y6Y1</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1077</td>\n",
       "      <td>Y/C</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681645</th>\n",
       "      <td>NC_000008.11:g.119105885C&gt;G</td>\n",
       "      <td>Q9Y6Z7</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>176</td>\n",
       "      <td>C/W</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31841 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  #Uploaded_variation   SWISSPROT REF_ALLELE Allele   \n",
       "9703        NC_000014.9:g.23104829G>A  A0A1B0GTW7          G      A  \\\n",
       "9922        NC_000014.9:g.23102250G>A  A0A1B0GTW7          G      A   \n",
       "14558     NC_000007.14:g.139161018A>G      A0AVF1          A      G   \n",
       "22222     NC_000007.14:g.122303281G>A      A0PJY2          G      A   \n",
       "23235      NC_000004.12:g.17526877C>A      A0PK11          C      A   \n",
       "...                               ...         ...        ...    ...   \n",
       "10678657   NC_000022.11:g.30937947C>A      Q9Y6X9          C      A   \n",
       "10678661   NC_000022.11:g.30937892G>A      Q9Y6X9          G      A   \n",
       "10680086    NC_000001.11:g.7677682C>T      Q9Y6Y1          C      T   \n",
       "10680187    NC_000001.11:g.7736507A>G      Q9Y6Y1          A      G   \n",
       "10681645  NC_000008.11:g.119105885C>G      Q9Y6Z7          C      G   \n",
       "\n",
       "               Consequence Protein_position Amino_acids   \n",
       "9703      missense_variant               31         S/F  \\\n",
       "9922      missense_variant              384         S/L   \n",
       "14558     missense_variant              263         N/S   \n",
       "22222     missense_variant              278         H/Y   \n",
       "23235     missense_variant              165         T/K   \n",
       "...                    ...              ...         ...   \n",
       "10678657  missense_variant              413         V/F   \n",
       "10678661  missense_variant              431         A/V   \n",
       "10680086  missense_variant              955         R/W   \n",
       "10680187  missense_variant             1077         Y/C   \n",
       "10681645  missense_variant              176         C/W   \n",
       "\n",
       "                              CLIN_SIG gnomADg_AF CANONICAL  \n",
       "9703                        pathogenic   0.001065       YES  \n",
       "9922                        pathogenic  1.971e-05       YES  \n",
       "14558     likely_pathogenic,pathogenic          -       YES  \n",
       "22222                       pathogenic          -       YES  \n",
       "23235                       pathogenic          -       YES  \n",
       "...                                ...        ...       ...  \n",
       "10678657             likely_pathogenic          -       YES  \n",
       "10678661  pathogenic,likely_pathogenic          -       YES  \n",
       "10680086             likely_pathogenic          -       YES  \n",
       "10680187             likely_pathogenic          -       YES  \n",
       "10681645                    pathogenic          -       YES  \n",
       "\n",
       "[31841 rows x 10 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "# remove redundancy\n",
    "import re\n",
    "\n",
    "def generate_key(tup):\n",
    "   return getattr(tup, 'SWISSPROT') + str(getattr(tup, 'Protein_position')) + getattr(tup, 'Amino_acids')\n",
    "\n",
    "def remove_redundancy(df):\n",
    "   encountered_dict = {}\n",
    "   boolMask = []\n",
    "   missenseRegex = r'(missense_variant)'\n",
    "\n",
    "   for row in df.itertuples():\n",
    "      key = generate_key(row)\n",
    "      consequence = getattr(row, 'Consequence')\n",
    "      try:\n",
    "         if encountered_dict[key] == True:\n",
    "            boolMask.append(False)\n",
    "      except KeyError:\n",
    "         encountered_dict[key] = True\n",
    "         if bool(re.search(consequence, missenseRegex)) == True:\n",
    "            boolMask.append(True)\n",
    "         else:\n",
    "            boolMask.append(False)\n",
    "\n",
    "   return df[boolMask]\n",
    "\n",
    "vep_benign_reviewed_df = remove_redundancy(vep_benign_reviewed_df)\n",
    "vep_pathogenic_reviewed_df = remove_redundancy(vep_pathogenic_reviewed_df)\n",
    "\n",
    "display(vep_benign_reviewed_df)\n",
    "display(vep_pathogenic_reviewed_df)\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Now we need to drop all the unnecessary columns and format `Amino_acids` column into `WT` and `Mut` columns**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>uniprot</th>\n",
       "      <th>position</th>\n",
       "      <th>WT</th>\n",
       "      <th>Mut</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>A0A075B6H7</td>\n",
       "      <td>39</td>\n",
       "      <td>V</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>A0A075B6H7</td>\n",
       "      <td>55</td>\n",
       "      <td>T</td>\n",
       "      <td>S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>A0A075B6H7</td>\n",
       "      <td>78</td>\n",
       "      <td>S</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>411</th>\n",
       "      <td>A0A075B6H8</td>\n",
       "      <td>27</td>\n",
       "      <td>T</td>\n",
       "      <td>I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>647</th>\n",
       "      <td>A0A075B6I3</td>\n",
       "      <td>61</td>\n",
       "      <td>L</td>\n",
       "      <td>P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681650</th>\n",
       "      <td>Q9Y6Z7</td>\n",
       "      <td>179</td>\n",
       "      <td>R</td>\n",
       "      <td>W</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681854</th>\n",
       "      <td>W5XKT8</td>\n",
       "      <td>246</td>\n",
       "      <td>G</td>\n",
       "      <td>S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681877</th>\n",
       "      <td>W5XKT8</td>\n",
       "      <td>265</td>\n",
       "      <td>W</td>\n",
       "      <td>R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681921</th>\n",
       "      <td>W6CW81</td>\n",
       "      <td>40</td>\n",
       "      <td>L</td>\n",
       "      <td>Q</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681932</th>\n",
       "      <td>W6CW81</td>\n",
       "      <td>56</td>\n",
       "      <td>H</td>\n",
       "      <td>R</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>58246 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             uniprot position WT Mut\n",
       "59        A0A075B6H7       39  V   A\n",
       "95        A0A075B6H7       55  T   S\n",
       "143       A0A075B6H7       78  S   G\n",
       "411       A0A075B6H8       27  T   I\n",
       "647       A0A075B6I3       61  L   P\n",
       "...              ...      ... ..  ..\n",
       "10681650      Q9Y6Z7      179  R   W\n",
       "10681854      W5XKT8      246  G   S\n",
       "10681877      W5XKT8      265  W   R\n",
       "10681921      W6CW81       40  L   Q\n",
       "10681932      W6CW81       56  H   R\n",
       "\n",
       "[58246 rows x 4 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>uniprot</th>\n",
       "      <th>position</th>\n",
       "      <th>WT</th>\n",
       "      <th>Mut</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9703</th>\n",
       "      <td>A0A1B0GTW7</td>\n",
       "      <td>31</td>\n",
       "      <td>S</td>\n",
       "      <td>F</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9922</th>\n",
       "      <td>A0A1B0GTW7</td>\n",
       "      <td>384</td>\n",
       "      <td>S</td>\n",
       "      <td>L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14558</th>\n",
       "      <td>A0AVF1</td>\n",
       "      <td>263</td>\n",
       "      <td>N</td>\n",
       "      <td>S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22222</th>\n",
       "      <td>A0PJY2</td>\n",
       "      <td>278</td>\n",
       "      <td>H</td>\n",
       "      <td>Y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23235</th>\n",
       "      <td>A0PK11</td>\n",
       "      <td>165</td>\n",
       "      <td>T</td>\n",
       "      <td>K</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10678657</th>\n",
       "      <td>Q9Y6X9</td>\n",
       "      <td>413</td>\n",
       "      <td>V</td>\n",
       "      <td>F</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10678661</th>\n",
       "      <td>Q9Y6X9</td>\n",
       "      <td>431</td>\n",
       "      <td>A</td>\n",
       "      <td>V</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10680086</th>\n",
       "      <td>Q9Y6Y1</td>\n",
       "      <td>955</td>\n",
       "      <td>R</td>\n",
       "      <td>W</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10680187</th>\n",
       "      <td>Q9Y6Y1</td>\n",
       "      <td>1077</td>\n",
       "      <td>Y</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10681645</th>\n",
       "      <td>Q9Y6Z7</td>\n",
       "      <td>176</td>\n",
       "      <td>C</td>\n",
       "      <td>W</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31841 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             uniprot position WT Mut\n",
       "9703      A0A1B0GTW7       31  S   F\n",
       "9922      A0A1B0GTW7      384  S   L\n",
       "14558         A0AVF1      263  N   S\n",
       "22222         A0PJY2      278  H   Y\n",
       "23235         A0PK11      165  T   K\n",
       "...              ...      ... ..  ..\n",
       "10678657      Q9Y6X9      413  V   F\n",
       "10678661      Q9Y6X9      431  A   V\n",
       "10680086      Q9Y6Y1      955  R   W\n",
       "10680187      Q9Y6Y1     1077  Y   C\n",
       "10681645      Q9Y6Z7      176  C   W\n",
       "\n",
       "[31841 rows x 4 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cols_to_drop = ['#Uploaded_variation', 'REF_ALLELE', 'Allele', 'Consequence', 'Amino_acids', 'CLIN_SIG', 'gnomADg_AF', 'CANONICAL']\n",
    "\n",
    "def get_WT(mutation):\n",
    "   \n",
    "   mutationStr = str(mutation)\n",
    "   \n",
    "   aa = mutationStr.split('/')\n",
    "   \n",
    "   return aa[0]\n",
    "\n",
    "def get_Mut(mutation):\n",
    "   \n",
    "   mutationStr = str(mutation)\n",
    "   \n",
    "   try:\n",
    "      aa = mutationStr.split('/')[1]\n",
    "      return aa\n",
    "   except IndexError:\n",
    "      return mutation\n",
    "\n",
    "# benign\n",
    "vep_benign_reviewed_df['WT'] = vep_benign_reviewed_df['Amino_acids'].apply(get_WT)\n",
    "vep_benign_reviewed_df['Mut'] = vep_benign_reviewed_df['Amino_acids'].apply(get_Mut)\n",
    "\n",
    "# pathogenic\n",
    "vep_pathogenic_reviewed_df['WT'] = vep_pathogenic_reviewed_df['Amino_acids'].apply(get_WT)\n",
    "vep_pathogenic_reviewed_df['Mut'] = vep_pathogenic_reviewed_df['Amino_acids'].apply(get_Mut)\n",
    "\n",
    "vep_benign_reviewed_df = vep_benign_reviewed_df.drop(columns=cols_to_drop).rename(columns={'SWISSPROT': 'uniprot', 'Protein_position': 'position'})\n",
    "vep_pathogenic_reviewed_df = vep_pathogenic_reviewed_df.drop(columns=cols_to_drop).rename(columns={'SWISSPROT': 'uniprot', 'Protein_position': 'position'})\n",
    "\n",
    "display(vep_benign_reviewed_df)\n",
    "display(vep_pathogenic_reviewed_df)\n",
    "\n",
    "OUTPUT_PATH = os.path.join(ROOT_DIR, 'synced_files/vep_variants')\n",
    "\n",
    "BENIGN_OUTFILE = os.path.join(OUTPUT_PATH, 'vep_benign_variants.csv')\n",
    "PATHOG_OUTFILE = os.path.join(OUTPUT_PATH, 'vep_pathogenic_variants.csv')\n",
    "\n",
    "vep_benign_reviewed_df.to_csv(BENIGN_OUTFILE, sep='\\t', index=False)\n",
    "vep_pathogenic_reviewed_df.to_csv(PATHOG_OUTFILE, sep='\\t', index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
